### PR TITLE
Work around various PyTorch issues in support of convolution.

### PR DESCRIPTION
**Pull request by stellaraccident**
* Enables the conv2d fwd test and ResA (which are both small).
* Deletes resnet18 and vgg, which both run but generate output that crashes FileCheck and lit (or at least makes them take an eternity).

### CODE REVIEW

File: frontends/pytorch/test/acap_export/test_export_add3.py [hunk 1]
```diff
@@ -13,6 +13,9 @@
 
 mb = torch_mlir.ModuleBuilder()
 with mb.capture_function("add3", [t0, t1, t2]) as f:
+  # t0 = f.import_tensor(t0)
```
Comment (silvasean) [inline frontends/pytorch/test/acap_export/test_export_add3.py:16]: what are these comments for?

File: frontends/pytorch/test/acap_export/test_export_conv2d_fwd.py [hunk 2]
```diff
@@ -29,13 +27,33 @@
 tensor = torch.randn(N, Cin, h, w)
 
 with mb.capture_function("conv2d_fwd", [tensor]) as f:
+  #tensor = f.import_tensor(tensor)
```
Comment (silvasean) [inline frontends/pytorch/test/acap_export/test_export_conv2d_fwd.py:30]: what is this line for?

File: frontends/pytorch/csrc/c10_dispatch/acap_dispatch.cpp [hunk 3]
```diff
@@ -34,9 +35,76 @@ using c10::Stack;
...
+  }
+
+  // Add to debug log.
+  std::stringstream sout;
+  sout << "CAPTURE: " << kernelName << "\n";
+  parent.captureLog.push_back(sout.str());
+  return op;
+}
+
+at::Tensor AcapController::importTensor(at::Tensor tensor) {
```
Comment (silvasean) [inline frontends/pytorch/csrc/c10_dispatch/acap_dispatch.cpp:100]: unclear what this function is doing. more comments?

File: frontends/pytorch/csrc/c10_dispatch/acap_dispatch.cpp [hunk 4]
```diff
@@ -34,9 +35,76 @@ using c10::Stack;
...
+  // TODO: Move the tensor to the CPU.
+  auto impl = c10::make_intrusive<at::TensorImpl>(
```
Comment (silvasean) [inline frontends/pytorch/csrc/c10_dispatch/acap_dispatch.cpp:102]: seems a little strange for our code to be manipulating an "impl" object. Comment?

File: frontends/pytorch/csrc/c10_dispatch/acap_dispatch.cpp [hunk 5]
```diff
@@ -48,8 +116,11 @@ py::object AcapController::contextEnter() {
   auto &stack = getThreadLocalActiveStack();
   stack.emplace_front(shared_from_this());
   Activation &current = stack.front();
-  current.dispatchGuard =
-      std::make_unique<c10::impl::IncludeDispatchKeyGuard>(kAcapDispatchKey);
+  c10::DispatchKeySet keySet{kAcapDispatchKey, kAcapGradDispatchKey};
+  current.includeGuard =
+      std::make_unique<c10::impl::IncludeDispatchKeyGuard>(keySet);
+  // current.excludeGuard =
```
Comment (silvasean) [inline frontends/pytorch/csrc/c10_dispatch/acap_dispatch.cpp:122]: remove commented code?

File: frontends/pytorch/csrc/c10_dispatch/acap_dispatch.cpp [hunk 6]
```diff
@@ -127,6 +198,66 @@ void AcapController::fallbackKernel(const OperatorHandle &opHandle,
...
+      const at::Tensor &, const at::Tensor &, const c10::optional<at::Tensor> &,
+      const at::IntArrayRef, const at::IntArrayRef, const at::IntArrayRef,
+      const bool, const at::IntArrayRef, const int64_t)>();
+
+  // Exclude recursive calls: convolution is completely emitted by this
+  // kernel.
+  c10::DispatchKeySet keySet{kAcapDispatchKey, kAcapGradDispatchKey};
+  c10::impl::ExcludeDispatchKeyGuard exclusion(keySet);
+
+  auto current = getCurrent();
```
Comment (silvasean) [inline frontends/pytorch/csrc/c10_dispatch/acap_dispatch.cpp:221]: nit: out of context, getCurrent() with the auto was kind of hard for me to grok. if possible, avoiding auto or renaming this to getCurrentThreadAcapController() or some such.

(I came into this with an understanding of the opengl "makeCurrent" terminology; and was still confused until I literally looked up what getCurrent returns here; call it bad memory of what the "current thing" is in this situation :P )

File: frontends/pytorch/csrc/c10_dispatch/acap_dispatch.cpp [hunk 7]
```diff
@@ -127,6 +198,66 @@ void AcapController::fallbackKernel(const OperatorHandle &opHandle,
...
+  if (!current) {
```
Comment (silvasean) [inline frontends/pytorch/csrc/c10_dispatch/acap_dispatch.cpp:222]: is there a deeper explanation of the `if (!current)` besides "we're just not capturing right now?".

**Review by silvasean — APPROVED** [review]
Looks great! Too bad about the convolution specialness :/

File: frontends/pytorch/test/acap_export/test_export_add3.py (hunk 1 above)
Comment (stellaraccident) [reply frontends/pytorch/test/acap_export/test_export_add3.py:16]: Oops - I thought I had backed out all of the hacking I had done to find the solution, but had forgotten about import_tensor. Just removed it entirely.

File: frontends/pytorch/test/acap_export/test_export_conv2d_fwd.py (hunk 2 above)
Comment (stellaraccident) [reply frontends/pytorch/test/acap_export/test_export_conv2d_fwd.py:30]: This is what I get for trying to recover and submit an original "tmp_hack_do_not_submit" branch :)

File: frontends/pytorch/csrc/c10_dispatch/acap_dispatch.cpp (hunk 6 above)
Comment (stellaraccident) [reply frontends/pytorch/csrc/c10_dispatch/acap_dispatch.cpp:221]: Renamed. The type is kind of gross.

File: frontends/pytorch/csrc/c10_dispatch/acap_dispatch.cpp (hunk 7 above)
Comment (stellaraccident) [reply frontends/pytorch/csrc/c10_dispatch/acap_dispatch.cpp:222]: That is the explanation. Left a comment on getCurrentThreadAcapController().

**Pull request merged**